File: components/sdk/dotnet/src/Spas.Sdk.Metadata/Builders/ConsistencyBuilder.cs
using Spas.Sdk.Metadata.Models;

namespace Spas.Sdk.Metadata.Builders;

/// <summary>
/// Builder for consistency metadata.
/// </summary>
public class ConsistencyBuilder
{
    private string _commands = "ACID";
    private string _queries = "EVENTUAL";

    /// <summary>
    /// Sets commands consistency (default: ACID).
    /// </summary>
    public ConsistencyBuilder WithCommands(string consistency)
    {
        _commands = consistency;
        return this;
    }

    /// <summary>
    /// Sets queries consistency (STRONG | EVENTUAL).
    /// </summary>
    public ConsistencyBuilder WithQueries(string consistency)
    {
        _queries = consistency;
        return this;
    }

    /// <summary>
    /// Builds the consistency metadata.
    /// </summary>
    public ConsistencyMetadata Build()
    {
        return new ConsistencyMetadata
        {
            Commands = _commands,
            Queries = _queries
        };
    }
}

File: components/sdk/java/spas-sdk-metadata/src/main/java/io/spas/sdk/metadata/builders/ConsistencyBuilder.java
package io.spas.sdk.metadata.builders;

import io.spas.sdk.metadata.model.Consistency;
import io.spas.sdk.metadata.model.ConsistencyLevel;
import io.spas.sdk.metadata.model.QueryConsistencyLevel;

/**
 * Fluent builder for consistency metadata.
 */
public final class ConsistencyBuilder {
    private ConsistencyLevel commands;
    private QueryConsistencyLevel queries;
    
    private ConsistencyBuilder() {}
    
    public static ConsistencyBuilder create() {
        return new ConsistencyBuilder();
    }
    
    public ConsistencyBuilder withCommands(ConsistencyLevel level) {
        this.commands = level;
        return this;
    }
    
    public ConsistencyBuilder withQueries(QueryConsistencyLevel level) {
        this.queries = level;
        return this;
    }
    
    public Consistency build() {
        return new Consistency(commands, queries);
    }
}
